File: 1/javascript/answer1.js
const { readFileSync } = require("fs");

const q = readFileSync("../question", { encoding: "utf-8" });
const lines = q.split("\n");

function findMatches(line) {
    const matches = line.match(/[0-9]/g);
    const firstMatch = matches.at(0);
    const lastMatch = matches.at(-1);

    console.log(firstMatch, lastMatch);
    return { firstMatch, lastMatch };
}

function solution(lines) {
    let sum = 0;

    lines.forEach((line) => {
        let { firstMatch, lastMatch } = findMatches(line);
        const match = firstMatch + lastMatch;

        sum += Number(match);
    });

    console.log(sum);
    return sum;
}

solution(lines);

File: 1/go/answer1.go
package main

import (
	"fmt"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func find_matches(line string) (n1 string, n2 string) {
	var m1 string = ""
	var m2 string = ""
	for i := 0; i < len(line); i++ {
		matches, err := regexp.MatchString(`[0-9]`, line[i:i+1])

		if err != nil {
			fmt.Println(err)
			return
		}

		if matches {
			if m1 == "" {
				m1 = line[i : i+1]
			}

			m2 = line[i : i+1]
		}

	}

	return m1, m2
}

func main() {
	data, err := os.ReadFile("../question")

	if err != nil {
		fmt.Println("Failed to read file")
		return
	}

	data_lines := strings.Split(string(data), "\n")

	var sum int64 = 0

	for _, line := range data_lines {
		n1, n2 := find_matches(line)

		cur_match, err := strconv.ParseInt((n1 + n2), 10, 64)
		if err != nil {
			fmt.Println("Failed to parse int")
			return
		}

		sum += cur_match
	}

	fmt.Println(sum)
}
